test/Search_UnitTestHelpers.hpp

```diff
@@ -144,58 +146,65 @@ zip(ExecutionSpace const &space, Kokkos::View<int *, DeviceType> indices,
 }
 
 #ifdef ARBORX_ENABLE_MPI
-template <typename Tree, typename Queries>
-auto query_with_distance(Tree const &tree, Queries const &queries,
+template <typename ExecutionSpace, typename Tree, typename Queries>
+auto query_with_distance(ExecutionSpace const &exec_space, Tree const &tree,
+                         Queries const &queries,
                          std::enable_if_t<is_distributed<Tree>{}> * = nullptr)
 {
-  using device_type = typename Tree::device_type;
-  Kokkos::View<int *, device_type> offsets("Testing::offsets", 0);
-  Kokkos::View<int *, device_type> indices("Testing::indices", 0);
-  Kokkos::View<int *, device_type> ranks("Testing::ranks", 0);
-  Kokkos::View<float *, device_type> distances("Testing::distances", 0);
-  using ExecutionSpace = typename device_type::execution_space;
-  ExecutionSpace space;
-  ArborX::Details::DistributedTreeImpl<device_type>::queryDispatchImpl(
-      ArborX::Details::NearestPredicateTag{}, tree, space, queries, indices,
-      offsets, ranks, &distances);
-
-  auto values = zip(space, indices, ranks, distances);
+  using MemorySpace = typename Tree::memory_space;
+  using DeviceType = Kokkos::Device<ExecutionSpace, MemorySpace>;
+
+  Kokkos::View<int *, MemorySpace> offsets("Testing::offsets", 0);
+  Kokkos::View<int *, MemorySpace> indices("Testing::indices", 0);
+  Kokkos::View<int *, MemorySpace> ranks("Testing::ranks", 0);
+  Kokkos::View<float *, MemorySpace> distances("Testing::distances", 0);
+  ArborX::Details::DistributedTreeImpl<DeviceType>::queryDispatchImpl(
+      ArborX::Details::NearestPredicateTag{}, tree, exec_space, queries,
+      indices, offsets, ranks, &distances);
+
+  auto values = zip(exec_space, indices, ranks, distances);
 
   return make_compressed_storage(
       Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offsets),
       Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, values));
 }
 #endif
 
-#define ARBORX_TEST_QUERY_TREE_WITH_DISTANCE(tree, queries, reference)         \
-  BOOST_TEST(query_with_distance(tree, queries) == (reference),                \
+#define ARBORX_TEST_QUERY_TREE_WITH_DISTANCE(exec_space, tree, queries,        \
+                                             reference)                        \
+  BOOST_TEST(query_with_distance(exec_space, tree, queries) == (reference),    \
              boost::test_tools::per_element());
 
-template <typename Tree>
-auto make(std::vector<ArborX::Box> const &b)
+template <typename ExecutionSpace, typename Tree>
```

Comment:
This is a mistake.  Switch the template parameter order so you can use FTAD `make<Tree>(exec_space)`

Reply:
Don't get it, need to learn more about FTAD.